proto/viam/app/data/v1/data.proto

```diff
@@ -80,6 +80,18 @@ message Filter {
   repeated string org_ids = 11;
   repeated string mime_type = 12;
   CaptureInterval interval = 13;
+  TagsFilterType tags_filter_type = 14;
```

Comment:
Only remaining change (and can do so in the first backwards-compatible way!) would be to instead have a TagsFilter submessage here since `tags` and `tags_filter_type` will always be paired together. Then LGTM to merge!

```
message TagsFilter {
  TagsFilterType type = 1;
  repeated string tags = 2;
}

message Filter {
 ..
 TagsFilter tags_filter = 14;
 repeated strings tags = 5 [deprecated=true];
}
```

In the future if we support more complicated queries, we can always add a new generic field to the above like `string query = 3`

We can later remove the deprecated field after updating app & RDK to use the new field